### ISSUE TITLE
Creating Records With Embedded Models

### ISSUE THREAD

**Issue by jamiechong** [firebase/emberfire#247]
I'm trying to implement my data structure similar to the discussion here: 
http://stackoverflow.com/questions/14296625/restricting-child-field-access-with-security-rules

The gist is that if I want to control read/write access to specific properties for a node, I must create a child node for those properties. 

So in the example above a naive approach is to create a Model like this: 

```
App.Entry = DS.Model.extend({
    //public
    name: DS.attr('string'),
    
    //private
    phone: DS.attr('string'),
    email: DS.attr('string'),
});
```

But now I can't lock down the private properties using Firebase's security rules. 

I can create a few one-to-one relationships and preferably embed the relations to make the security rules in Firebase easy to write.

```
App.Entry = DS.Model.extend({
    public: DS.belongsTo('publicEntry', {embedded: true}),
    private: DS.belongsTo('privateEntry', {embedded: true}),
});

App.PublicEntry = DS.Model.extend({
    name: DS.attr('string')
});

App.PrivateEntry = DS.Model.extend({
    phone: DS.attr('string'),
    email: DS.attr('string'),
});
```

The structure I'd see in Firebase would be: 

```
-- entries
   -- entry1
      -- public
          -- name: 'Jamie'
      -- private
         -- email: 'jamie@email.com'
         -- phone: '555 1212'
```
   

But now I'm getting hung out to dry when I try to create these  records. 

```
this.store.createRecord('entry', {
    public: this.store.createRecord('publicEntry', {
        name: 'Jamie'
    },
    private: this.store.createRecord('privateEntry', {
        email: 'jamie@email.com',
        phone: '555 1212'
    }
}).save();
```

This seems to work - it creates the desirable data structure in Firebase, but I'm getting an error about the payload not having the correct id

`Error: Embedded relationship "publicEntry" of "app@model:entry:" must contain an "id" property in the payload`

I've seen some of your other comments about how `embedded` relationships are buggy and not recommended and `async` relationships are preferable. However for me to be able to write the FIrebase security rules that I want, I don't see how async is going to work. 

Can you offer a suggestion for a pattern that would work with EmberFire? 

Also note that the example above isn't the actual code I'm trying to write. In reality my situation is even more complex. I essentially I have two users with difference access levels writing to the same record. I don't want User A to be able to write over User B's properties. 

Just wanted to open this issue up for discussion. Thanks!

**Comment by tstirrat** [comment]
What you are trying to do should work. As in #200 there is a bug where embedded records are not being given an embedded id.

**Comment by tstirrat** [comment]
#200 contains a workaround where you can create a custom serializer for your embedded types and force them to serialize their `id` property

**Comment by tstirrat** [comment]
#249 contains some updates that should fix this.

#138 shows you how to test this in your app.

**Comment by tstirrat** [comment]
closing this as a duplicate of 200, so if you have any more questions, lets talk in 200

**Comment by tstirrat** [comment]
Fixed in #249
